src/components/accounts/two_factor/EnableTwoFactor.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { Translate } from 'react-localize-redux';
+import TwoFactorOption from './TwoFactorOption';
+import PhoneInput, { isValidPhoneNumber } from 'react-phone-number-input';
+import { validateEmail } from '../../../utils/account';
+import FormButton from '../../common/FormButton';
+import {
+    initTwoFactor,
+    reInitTwoFactor,
+    verifyTwoFactor,
+    deployMultisig,
+    redirectToApp
+} from '../../../actions/account';
+import { useRecoveryMethods } from '../../../hooks/recoveryMethods';
+import EnterVerificationCode from '../EnterVerificationCode'
+import Container from '../../common/styled/Container.css'
+
+const StyledContainer = styled(Container)`
+
+    h4 {
+        margin-top: 40px;
+
+        span {
+            color: #FF585D;
+        }
+    }
+
+    button {
+        margin-top: 50px !important;
+        width: 100% !important;
+    }
+`
+
+export function EnableTwoFactor(props) {
+
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const accountId = account.accountId;
+
+    const [initiated, setInitiated] = useState(false);
+    const [option, setOption] = useState('email');
+    const [email, setEmail] = useState('');
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const recoveryMethods = useRecoveryMethods(accountId);
+    const loading = account.actionsPending.some(action => ['INIT_TWO_FACTOR', 'VERIFY_TWO_FACTOR', 'DEPLOY_MULTISIG'].includes(action))
+
+    const method = {
+        kind: `2fa-${option}`,
+        detail: option === 'email' ? email : phoneNumber
+    }
+
+    useEffect(() => { 
+        const email = recoveryMethods.filter(method => method.kind === 'email')[0];
+        const phone = recoveryMethods.filter(method => method.kind === 'phone')[0];
+
+        if (email) {
+            setEmail(email.detail)
+        }
+
+        if (phone) {
+            setPhoneNumber(phone.detail)
+        }
+        
+    }, [recoveryMethods]);
+
+    const handleNext = async () => {
+
+        const { confirmed, error } = await dispatch(initTwoFactor(accountId, method))
```

Comment:
see https://github.com/near/near-wallet/pull/671, error handling for redux actions doesn't work like this anymore

which makes me think – was error case code tested anyhow?